viz/cmd/install.go

```diff
@@ -62,10 +61,11 @@ func newCmdInstall() *cobra.Command {
 The installation can be configured by using the --set, --values, --set-string and --set-file flags.
 A full list of configurable values can be found at https://www.github.com/linkerd/linkerd2/tree/main/viz/charts/linkerd-viz/README.md
   `,
-		RunE: func(cmd *cobra.Command, args []string) error {
+		RunE: func(_ *cobra.Command, _ []string) error {
+			cniEnabled := false
```

Comment:
We could also declare this at the top where we have  `ha`, `skipChecks` and `ignoreCluster` instead of instantiating here. In a sense, we already do this with `ha` so we could probably be consistent with that pattern.

We can still pass it as an argument to the function, I think that makes it more readable.

Reply:
I'll move it to the top as you suggest.

> We can still pass it as an argument to the function, I think that makes it more readable.

Can you please clarify?

Reply:
Yeah, ofc! Could have been a bit more clearer.

In places like `cli/cmd/install.go` we actually declare these values (e.g `ignoreCluster`) in the outer scope, not in the enclosing function's scope (the command builder function, `newCmdInstall`). If we have global state, we can access it (afaik) from other functions without having to pass it in as an argument.

https://github.com/linkerd/linkerd2/blob/26dbd4e063f07d1ce627ecbb63e3943cd707738b/cli/cmd/install.go#L70-L72

I like the approach here more because we don't rely on global state. Passing the values in as an argument to the function `install(w io.Writer, options values.Options, ha, cniEnabled bool)` also reads better to me.

Reply:
@mateiidavid ok cool, so I think my last commit addresses all your feedback :wink: